src/Components/Replication/ReplicationSettingsForm.tsx

```diff
@@ -102,6 +117,73 @@ export default class ReplicationSettingsFormReplication extends React.Component<
     }
 
     try {
+      let parsedWhen: Record<string, any> | undefined;
```

Comment:
You should not be checking operators and labels here, only JSON syntax. The database will do that for you. It is a bad idea to let the client application know which operators are supported. Because we have to update it every time we add a new operator.